resources/js/views/settings/roles/View.vue

```diff
@@ -0,0 +1,294 @@
+          </b-col>
+        </b-row>
+      </b-container>
+    </b-form>
+
+    <b-modal
+      :static='modalStatic'
+      :busy='isBusy'
+      ok-variant='danger'
+      cancel-variant='dark'
+      @ok='forceOverwrite'
+      @cancel='refreshRole'
+      :hide-header-close='true'
+      :no-close-on-backdrop='true'
+      :no-close-on-esc='true'
+      ref='stale-role-modal'
+      :hide-header='true'>
+      <template v-slot:default>
+        <h5>{{ staleError.message }}</h5>
+      </template>
+      <template v-slot:modal-ok>
+        <b-spinner small v-if="isBusy"></b-spinner>  {{ $t('app.overwrite') }}
+      </template>
+      <template v-slot:modal-cancel>
+        <b-spinner small v-if="isBusy"></b-spinner>  {{ $t('app.reload') }}
+      </template>
+    </b-modal>
+  </div>
+</template>
+
+<script>
+import Base from '../../../api/base';
+import FieldErrors from '../../../mixins/FieldErrors';
+import { mapGetters } from 'vuex';
+import env from '../../../env';
+
+export default {
+  mixins: [FieldErrors],
+
+  props: {
+    id: {
+      type: [String, Number],
+      required: true
+    },
+
+    viewOnly: {
+      type: Boolean,
+      required: true
+    },
+
+    modalStatic: {
+      type: Boolean,
+      default: false
+    }
+  },
+
+  computed: {
+    ...mapGetters({
+      settings: 'session/settings'
+    })
+  },
+
+  data () {
+    return {
+      isBusy: false,
+      model: {},
+      permissions: {},
+      permissionsLoading: false,
+      errors: {},
+      staleError: {}
+    };
+  },
+
+  /**
+   * Loads the role from the backend and also a part of permissions that can be selected.
+   */
+  mounted () {
+    this.loadPermissions();
+
+    if (this.id !== 'new') {
+      this.isBusy = true;
+
+      Base.call(`roles/${this.id}`).then(response => {
+        this.model = response.data.data;
+        this.model.permissions = this.model.permissions.map(permission => permission.id);
+      }).catch(response => {
+        Base.error(response, this.$root, response.message);
+      }).finally(() => {
+        this.isBusy = false;
+      });
+    }
+  },
+
+  methods: {
+    /**
+     * Loads the permissions that can be selected through checkboxes.
+     */
+    loadPermissions () {
+      this.permissionsLoading = true;
+
+      Base.call('permissions').then(response => {
+        this.permissions = {};
+        response.data.data.forEach(permission => {
+          const group = permission.name.split('.')[0];
+
+          if (!this.permissions[group]) {
+            this.permissions[group] = [];
+          }
+
+          permission.translatedName = this.$t(`app.permissions.${permission.name}`);
+          this.permissions[group].push(permission);
+        });
+      }).catch(error => {
+        Base.error(error, this.$root, error.message);
+      }).finally(() => {
+        this.permissionsLoading = false;
+      });
+    },
+
+    /**
+     * Goes back to the role overview page.
+     */
+    back () {
```

Comment:
Whats  the purpose of this method? Why not do it inline?

Reply:
Done